geopandas/tests/test_op_output_types.py

```diff
@@ -271,3 +288,58 @@ def test_expanddim_in_unstack():
         assert unstack._geometry_column_name is None
     else:  # pandas GH37369, unstack doesn't call finalize
         assert unstack._geometry_column_name == "geometry"
+
+
+# indexing /  constructor_sliced tests
+
+test_case_column_sets = [
+    ["geometry"],
+    ["geometry2"],
+    ["geometry", "geometry2"],
+    # non active geo col case
+    ["geometry", "value"],
+    ["geometry", "value_nan"],
+    ["geometry2", "value"],
+    ["geometry2", "value_nan"],
+]
+
+
+@pytest.mark.parametrize(
+    "column_set",
+    test_case_column_sets,
+    ids=[", ".join(i) for i in test_case_column_sets],
+)
+def test_constructor_sliced_row_slices(df2, column_set):
+    # https://github.com/geopandas/geopandas/issues/2282
+    df_subset = df2[column_set]
+    assert isinstance(df_subset, pd.DataFrame)
```

Comment:
```suggestion
    assert isinstance(df_subset, GeoDataFrame)
```

All those cases should also be a GeoDataFrame here?

Reply:
Good catch - you're quite right (aside, this doesn't pass with the suggestion as written because my test df was bad, `df2` has `geometry2` and `geometry3` as lists of shapely objects, which I suppose is how I missed this, and this preceded the getitem changes)